backend/db/migrate/20220426073907_create_favourite_projects.rb

```diff
@@ -0,0 +1,12 @@
+class CreateFavouriteProjects < ActiveRecord::Migration[7.0]
+  def change
+    create_table :favourite_projects, id: :uuid do |t|
+      t.belongs_to :user, foreign_key: {on_delete: :cascade}, type: :uuid, null: false
+      t.belongs_to :project, foreign_key: {on_delete: :cascade}, type: :uuid, null: false
+
+      t.index [:user_id, :project_id], unique: true
```

Comment:
do we have anything to prevent hitting pg uniqueness error in case we get request to create a duplicate favourite? what response would FE get in such case?

Reply:
Favourite record is created via `@project.favourite_projects.create! user: current_user` command. What do you think about changing `create!` to `create` so when creation of record fails because of duplicity, it will not crash, but return valid response, which is in this case serialized Project object (`render json: ProjectSerializer.new(@project, params: {current_user: current_user}).serializable_hash`)?

Reply:
Ofc I can also add validation to model, just to be on save side. When duplicity occurs, we either return validation error or fails silently (project is already favourited so we return it, no need to return error)

Reply:
Thanks!